Add new blog post (#2846)
OWASP/owasp-mastg: docs/news/posts/2024-07-30-new-maswe.md

<center>
<img style="width: 40%; border-radius: 5px" src="/assets/maswe-overview.png"/>
</center>

## Refactoring the MASTG

We began the refactoring process in 2021, focusing first on the MASVS and then on the MASTG. Our primary goal was to break the MASTG v1 into modular components, including tests, techniques, tools, and applications.

<center>
<img style="width: 80%; border-radius: 5px" src="/assets/news/mas_v2_arch.png"/>
</center>

This modular approach allows us to **maintain and update each component independently**, ensuring that the MASTG remains current and relevant. For example, in our previous structure, the MASTG consisted of large test cases within a single markdown file. This was not only difficult to maintain but also made it challenging to reference specific tests; and it was impossible to have metadata for each test.

> The new structure divides tests into individual pages (Markdown files with metadata), each with its own ID (`MASTG-TEST-****`) and links to relevant techniques (`MASTG-TECH-****`) and tools (`MASTG-TOOL-****`). This encapsulation ensures that **each test is easily referenced and promotes reusability** across all MAS components. For example, you can now see which tools are used in a particular test and which techniques are applied. This is currently partially implemented, but when we're done, you'll be able to do many interesting things, like see all tests that use a particular tool or technique.

Reviewer: ```suggestion
The new structure divides tests into individual pages (Markdown files with metadata), each with its own ID (`MASTG-TEST-****`) and links to relevant techniques (`MASTG-TECH-****`) and tools (`MASTG-TOOL-****`). This encapsulation ensures that **each test is easily referenced and promotes reusability** across all MAS components. For example, you can now see which tools are used in a particular test and which techniques are applied. 
```
Delete this line? Feels very similar to the line before it.

> The new structure divides tests into individual pages (Markdown files with metadata), each with its own ID (`MASTG-TEST-****`) and links to relevant techniques (`MASTG-TECH-****`) and tools (`MASTG-TOOL-****`). This encapsulation ensures that **each test is easily referenced and promotes reusability** across all MAS components. For example, you can now see which tools are used in a particular test and which techniques are applied. This is currently partially implemented, but when we're done, you'll be able to do many interesting things, like see all tests that use a particular tool or technique.

Author: Second try. Hopefully it will be clearer this time.

```suggestion
The new structure divides tests into individual pages (Markdown files with metadata), each with its own ID (`MASTG-TEST-****`) and links to relevant techniques (`MASTG-TECH-****`) and tools (`MASTG-TOOL-****`). This encapsulation ensures that **each test is easily referenced and promotes reusability** across all MAS components. For example, you can open a test and see what tools and techniques are being used, and soon you'll be able to do the same in reverse: open a tool or technique and see all the tests that use it. This deep cross-referencing can be extremely powerful when exploring the MASTG.
```

<div style="display: flex; justify-content: center; align-items: center;">
    <div style="flex: 1; display: flex; justify-content: center; align-items: center;">
        <img style="width: 45%; border-radius: 5px" src="/assets/maswe_cover.png"/>
    </div>
    <div style="flex: 1; display: flex; justify-content: center; align-items: center;">
        <img style="width: 100%; border-radius: 5px" src="/assets/news/maswe_main.png"/>
    </div>
</div>

Now MASVS, MASWE and MASTG are all seamlessly connected. We start with the high-level requirements, zoom in on the specific weaknesses, and then go low-level to the tests and hands-on with the demos. Here's how it works:

1. **MASVS Controls**: High-level platform-agnostic requirements.
   > For example, "The app employs current cryptography and uses it according to best practices." ([MASVS-CRYPTO-1](https://mas.owasp.org/MASVS/controls/MASVS-CRYPTO-1/)).
2. **MASWE Weaknesses**: Specific weaknesses, typically also platform-agnostic, related to the controls.
   > For example, "use of weak pseudo-random number generation" ([MASWE-0027](https://mas.owasp.org/MASWE/MASVS-CRYPTO/MASWE-0027/)).

> 3. **MASTG Tests**: Each weakness is implemented by multiple tests that guide the tester in identifying and mitigating the issues using various tools and techniques on each mobile platform.

Reviewer: ```suggestion
3. **MASTG Tests**: Each weakness is evaluated by executing tests that guide the tester in identifying and mitigating the issues using various tools and techniques on each mobile platform.
```

>    > For example, testing for "insecure random API usage on Android" ([MASTG-TEST-0204](hhttps://mas.owasp.org/MASTG/tests-beta/android/MASVS-CRYPTO/MASTG-TEST-0204/)).

Author: ```suggestion
   > For example, testing for "insecure random API usage on Android" ([MASTG-TEST-0204](https://mas.owasp.org/MASTG/tests-beta/android/MASVS-CRYPTO/MASTG-TEST-0204/)).
```

> 4. **MASTG Demos**: Practical demonstrations that include working code samples and test scripts to ensure reproducibility and reliability.

Reviewer: ```suggestion
4. **MASTG Demos**: Practical demonstrations that include insecure code samples and test scripts to ensure reproducibility and reliability.
```

Author: The demos can also include secure code
